python/src/aac/lang/lsp/language_server.py

```diff
@@ -45,16 +51,53 @@ def configure_lsp(self):
         self.code_completion_provider = self.code_completion_provider or CodeCompletionProvider()
         self.setup_features()
 
-        logging.debug("AaC LSP initialized.")
-
     def setup_features(self) -> None:
         """Configure the server with the supported features."""
         server = self.language_server
+        server.sync_kind = TextDocumentSyncKind.FULL
+
+        @server.feature(methods.TEXT_DOCUMENT_DID_OPEN)
+        async def did_open(ls, params: DidOpenTextDocumentParams):
+            """Text document did open notification."""
+            logging.info(f"Text document opened by LSP client {params.text_document.uri}.")
+
+            file_uri = params.text_document.uri
+            managed_file = self.workspace_files.get(file_uri)
+
+            if not managed_file:
+                managed_file = ManagedWorkspaceFile(file_uri)
+                self.workspace_files[file_uri] = managed_file
+
+            managed_file.is_client_managed = True
+            _, file_path = file_uri.split("file://")
+            self.language_context.add_definitions_to_context(parse(file_path))
+
+        @server.feature(methods.TEXT_DOCUMENT_DID_CLOSE)
+        def did_close(server: LanguageServer, params: DidCloseTextDocumentParams):
+            """Text document did close notification."""
+            logging.info(f"Text document closed by LSP client {params.text_document.uri}.")
+
+            file_uri = params.text_document.uri
+            managed_file = self.workspace_files.get(file_uri)
+            managed_file.is_client_managed = False
+
+        @server.feature(methods.TEXT_DOCUMENT_DID_CHANGE)
+        def did_change(ls, params: DidChangeTextDocumentParams):
+            """Text document did change notification."""
+            logging.info(f"Text document altered by LSP client {params.text_document.uri}.")
+
+            file_content = params.content_changes[0].text
+            altered_definitions = parse(file_content)
+
+            for altered_definition in altered_definitions:
+                # At the moment we have to rely on definition names, but we'll need to update definitions based on file URI
```

Comment:
I've also identified some more blockers to properly implementing updates to the LSP. I've documented two of the issues here:
https://github.com/jondavid-black/AaC/issues/323
https://github.com/jondavid-black/AaC/issues/324